packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -608,35 +612,74 @@ class FlutterDevice {
     @required String dillOutputPath,
     @required List<Uri> invalidatedFiles,
     @required PackageConfig packageConfig,
+    @required WidgetCache widgetCache,
   }) async {
     final Status devFSStatus = globals.logger.startProgress(
       'Syncing files to device ${device.name}...',
       timeout: timeoutConfiguration.fastOperation,
     );
+
+    bool fastReassemble = false;
+    Future<void> attemptFastReassembleCheck() async {
+      final List<FlutterView> views = await vmService.getFlutterViews();
+      if (widgetCache == null || invalidatedFiles.length != 1) {
+        return;
+      }
+      final String widgetName = await widgetCache?.validateLibrary(invalidatedFiles.single);
+      if (widgetName == null) {
+        return;
+      }
+      for (final FlutterView view in views) {
+        final vm_service.Isolate isolate = await vmService.getIsolateOrNull(view.uiIsolate.id);
+        vm_service.LibraryRef targetLibrary;
+        for (final vm_service.LibraryRef libraryRef in isolate.libraries) {
+          if (libraryRef.uri == 'package:flutter_gallery/gallery/home.dart') {
+            targetLibrary = libraryRef;
+            break;
+          }
+        }
+        try {
+        await vmService.evaluate(
+          view.uiIsolate.id,
+          targetLibrary.id,
+          '((){debugFastReassembleMethod=(Object _) => _ is $widgetName})()',
```

Comment:
Should `debugFastReassembleMethod` get set back to `null` at some point?

Reply:
That is probably the safest choice

Reply:
actually if we are repeatedly changing the same class, then leaving this set would let us skip the expression evaluation changes. I'll look into whether this causes any issues